### ISSUE TITLE
Clarification on pretrained weights and _base_ config usage in Mask R-CNN

### ISSUE THREAD

**Issue by ZahraDalvand** [3dv-casia/PolyRoom#5]
Dear authors,

Thank you for providing the implementation of PolyRoom. While following the instructions to run the project, we encountered a few ambiguities and would appreciate your clarification.
In the file engine.py, we noticed the following two lines:
maskrcnnconfig_file = ''
maskrcnncheckpoint_file = ''
It is necessary to specify both the config file and the checkpoint file for the Mask R-CNN model. We identified that the config file is:

**mask-rcnn_r50-caffe_fpn_ms-poly-1x_str3d.py**

Which is already present in the repository next to main.py.

However, we could not find the corresponding pretrained weights (checkpoint) file in this repository. Based on the filename and context, we assumed that the correct pretrained weights file is:

**mask_rcnn_r50_fpn_1x_coco_20200205-d4b0c5d6.pth**
**We downloaded this file from the official MMDetection model zoo using the following link:**

https://download.openmmlab.com/mmdetection/v2.0/mask_rcnn/mask_rcnn_r50_fpn_1x_coco/mask_rcnn_r50_fpn_1x_coco_20200205-d4b0c5d6.pth

and placed it inside a checkpoint/ directory.
In addition, we noticed that the config file mask-rcnn_r50-caffe_fpn_ms-poly-1x_str3d.py includes references to several _base_ components. Therefore, **we downloaded the complete _base_ folder from the MMDetection GitHub repository:**
https://github.com/open-mmlab/mmdetection/tree/main/configs/_base_
and placed it in the same directory as the aforementioned config file.

We also enabled and executed the evaluate_floor() function to evaluate the model on the validation set. However, the predictions stored during evaluation (after 50 epochs of training) do not align well with the ground truth masks. The results appear suboptimal and do not converge satisfactorily.
**We would appreciate it if you could help us verify the following:
1- Have we downloaded and used the correct pretrained Mask R-CNN weights?
2- the _base_ folder that we used from the MMDetection repo compatible with this setup?
3-If not, could you please share the correct links to the expected checkpoint and base files?
4- Finally, in your experience, what is the minimum number of epochs typically  required for PolyRoom to produce reliable and high-quality predictions?**
Your insights would be extremely helpful for our experiments and further development.
Thank you very much for your time and for your excellent work on this project.
**Best regards**

**Comment by ZahraDalvand** [comment]
Thank you very much for your helpful response. My issue was resolved using the documentation link you provided:
https://mmdetection.readthedocs.io/en/latest/user_guides/train.html.

Also, I have successfully downloaded the pretrained Mask2Former.pth model and its configuration file (mask2former_config.py) from the following link:
https://drive.google.com/drive/folders/186SLocs6jPzCKq9B5419HfMV2TIMBz2I

Now, I can train the PolyRoom model on my training data and evaluate it on the validation dataset using the evaluate_floor function. This function saves the model's predictions as visual outputs in the output folder.

However, I noticed that after several epochs of training, the PolyRoom model fails to detect any polygons for any of the validation images — the prediction results are completely empty.
In contrast, the Mask R-CNN model detects several polygons per image under the same conditions.

I'm trying to understand why the PolyRoom model fails to learn in this case. One thing I’m wondering about is this: during training, I use train.json and val.json files (which contain annotations for all training and validation images, respectively), but the evaluate_floor function internally calls a load_annotation function that seems to load a separate annotation_3d.json file per scene.

Could this mismatch in annotation format between training and evaluation be causing the issue? Or is there possibly something wrong with how I am training the model?

Any guidance would be greatly appreciated.
Thanks again!

**Comment by PaulLiuYZ** [comment]
I wonder if the loss drops rightly, PolyRoom is based on the segmentation results, so that this is unusual, what's the threhold of corner seletction? 0.01 may be ok while 0.5 too large

**Comment by ZahraDalvand** [comment]
Yes, the loss values decrease during training. 
You were right — I had set the corner selection threshold to 0.5, which caused the output to have no polygons at all. After lowering the threshold to 0.1, the results improved significantly. 
**Thank you very much for your guidance and support. 
Wish you all the best!**
